pageserver/src/tenant.rs

```diff
@@ -882,28 +881,19 @@ impl Tenant {
                     // Remote deletion did not finish. Need to resume.
                     timelines_to_resume_deletion.push((
                         timeline_id,
-                        load_metadata(self.conf, &self.tenant_id, &timeline_id)?,
+                        Some(load_metadata(self.conf, &self.tenant_id, &timeline_id)?),
```

Comment:
On this path we do an exists check... Feels like we could fold these together like:

```
// this probably has some combinator I don't remember
let metadata = match load_metadata(self.conf, &self.tenant_id, &timeline_id) {
  Ok(m) => Some(m),
  Err(e) if e.kind() == std::io::ErrorKind::NotFound => None,
  Err(e) => return Err(e.into()),
};

timelines_to_resume_deletion.push((timeline_id, metadata));
```

Perhaps, what do you think? I think the less `exists()` checks we have, the better, because it's always a TOCTOU possibility and cases which fall between `exists()` and `try_exists()`. However I wouldn't want to recommend `try_exists()` because embracing that would still leave us TOCTOU possibility.

Of course exists needs to be used in *this* function but the less we have the better.

Reply:
Done